File: BE/HDDT_BE/Models/Professional/ThongTinPhatHanhModel.cs
using HDDT_BE.Models.Core;
using MongoDB.Bson.Serialization.Attributes;

namespace HDDT_BE.Models.Professional;


public class ThongTinPhatHanhModel : Audit, TEntity<string>
{
    
public string invoiceType {get; set;} // Mã loại hoá đơn

public string templateCode {get; set;} // Ký hiệu MẪU hoá đơn

public string invoiceSeries {get; set;} // Ký hiệu hoá đơn

public string currencyCode {get; set;} // Mã tiền tệ

public string adjustmentType  {get; set;} //Trạng thái điều chỉnh hóa đơn

public bool paymentStatus {get; set;} // 1 - Đã thanh toán - 2 chưa thanh toán

public bool cusGetInvoiceRight {get; set;} // Cho phép người dùng tra cứu hóa đơn hay không. MẶC ĐỊNH BẰNG TRUE
    
}




public class ThongTinPhatHanhModelShort{

    public string invoiceType {get; set;} // Mã loại hoá đơn
    public string templateCode {get; set;} // Ký hiệu MẪU hoá đơn

    public string invoiceSeries {get; set;} // Ký hiệu hoá đơn

    public string currencyCode {get; set;} // Mã tiền tệ
    
    public string adjustmentType  {get; set;} //Trạng thái điều chỉnh hóa đơn

    public bool paymentStatus {get; set;} // 1 - Đã thanh toán - 2 chưa thanh toán

    public bool cusGetInvoiceRight {get; set;} // Cho phép người dùng tra cứu hóa đơn hay không. MẶC ĐỊNH BẰNG TRUE
    
}



public class ThongTinPhatHanhCTModelShort{

    public string invoiceType {get; set;} // Mã loại hoá đơn
    public string templateCode {get; set;} // Ký hiệu MẪU hoá đơn

    public string invoiceSeries {get; set;} // Ký hiệu hoá đơn

    public string currencyCode {get; set;} // Mã tiền tệ

    public double invoiceIssuedDate { get; set; }

    public string transactionUuid { get; set; }
    public string adjustmentType  {get; set;} //Trạng thái điều chỉnh hóa đơn

    public bool paymentStatus {get; set;} // 1 - Đã thanh toán - 2 chưa thanh toán

    public bool cusGetInvoiceRight {get; set;} // Cho phép người dùng tra cứu hóa đơn hay không. MẶC ĐỊNH BẰNG TRUE
    
}



public class ThongTinPhatHanhTTModelShort{

    
    public string transactionUuid { get; set; }
    public string invoiceType {get; set;} // Mã loại hoá đơn
    public string templateCode {get; set;} // Ký hiệu MẪU hoá đơn

    public string invoiceSeries {get; set;} // Ký hiệu hoá đơn
    public string currencyCode {get; set;} // Mã tiền tệ
    public string adjustmentType  {get; set;} //Trạng thái điều chỉnh hóa đơn
    
    public string originalInvoiceId { get; set; }
    
    
     public double invoiceIssuedDate { get; set; }
     /*public DateTime originalInvoiceIssueDate { get; set; }
     
     public DateTime additionalReferenceDate { get; set; }*/
    

    public string additionalReferenceDesc { get; set; } = "VAN BAN";
    
    public bool paymentStatus {get; set;} // 1 - Đã thanh toán - 2 chưa thanh toán

     public bool cusGetInvoiceRight {get; set;} // Cho phép người dùng tra cứu hóa đơn hay không. MẶC ĐỊNH BẰNG TRUE
    
}

File: FE/src/models/thongTinPhatHanhModel.js
const toJson = (item) => {
    return {
        id: item.id,
        invoiceType: item.invoiceType,
        templateCode: item.templateCode,
        invoiceSeries: item.invoiceSeries,
        currencyCode : item.currencyCode,
        adjustmentType : item.adjustmentType,
        paymentStatus : item.paymentStatus ,
        cusGetInvoiceRight : item.cusGetInvoiceRight
    }
}

const fromJson = (item) => {
    return {
        id: item.id,
        invoiceType: item.invoiceType,
        templateCode: item.templateCode,
        invoiceSeries: item.invoiceSeries,
        currencyCode : item.currencyCode,
        adjustmentType : item.adjustmentType,
        paymentStatus : item.paymentStatus ,
        cusGetInvoiceRight : item.cusGetInvoiceRight
    }
}
const baseJson = () => {
    return {
        id: null,
        invoiceType: null,
        templateCode: null,
        invoiceSeries: null,
        currencyCode : null,
        adjustmentType : null,
        paymentStatus : true ,
        cusGetInvoiceRight : true
    }
}

const toListModel = (items) =>{
    if(items.length > 0){
        let data = [];
        items.map((value, index) =>{
            data.push(fromJson(value));
        })
        return data??[];
    }
    return [];
}
export const thongTinPhatHanhModel = {
    toJson, fromJson, baseJson, toListModel
}
